WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationStep.kt

```diff
@@ -1,34 +1,49 @@
 package org.wordpress.android.ui.sitecreation
 
+import org.wordpress.android.fluxc.model.experiments.Variation.Control
 import org.wordpress.android.ui.sitecreation.SiteCreationStep.DOMAINS
 import org.wordpress.android.ui.sitecreation.SiteCreationStep.INTENTS
-import org.wordpress.android.ui.sitecreation.SiteCreationStep.SEGMENTS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SITE_DESIGNS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SITE_NAME
 import org.wordpress.android.ui.sitecreation.SiteCreationStep.SITE_PREVIEW
 import org.wordpress.android.util.config.SiteIntentQuestionFeatureConfig
+import org.wordpress.android.util.config.SiteNameFeatureConfig
+import org.wordpress.android.util.experiments.SiteNameABExperiment
 import org.wordpress.android.util.wizard.WizardStep
 import javax.inject.Inject
 import javax.inject.Singleton
 
 enum class SiteCreationStep : WizardStep {
-    SEGMENTS, DOMAINS, SITE_PREVIEW, INTENTS;
+    SITE_DESIGNS, DOMAINS, SITE_PREVIEW, INTENTS, SITE_NAME;
 }
 
 @Singleton
 class SiteCreationStepsProvider @Inject constructor(
-    private val siteIntentQuestionFeatureConfig: SiteIntentQuestionFeatureConfig
+    private val siteIntentQuestionFeatureConfig: SiteIntentQuestionFeatureConfig,
+    private val siteNameFeatureConfig: SiteNameFeatureConfig,
+    private val siteNameABExperiment: SiteNameABExperiment,
 ) {
     fun getSteps(): List<SiteCreationStep> {
         if (siteIntentQuestionFeatureConfig.isEnabled()) {
+             if (siteNameFeatureConfig.isEnabled() && siteNameABExperiment.getVariation() != Control) {
```

Comment:
I wonder if we can keep the `siteIntentQuestionFeatureConfig` independent of the `siteNameFeatureConfig`. This is not blocking for this PR but maybe it will be more versatile if we composed the list more dynamically (e.g. add items to a mutable list when a dynamic feature is available)

Reply:
Since `Detekt` also complains about `ReturnCount` I was thinking of something like
```
    fun getSteps() = mutableListOf<SiteCreationStep>().apply {
        if (siteIntentQuestionFeatureConfig.isEnabled()) {
            add(INTENTS)
        }
        if (siteNameFeatureConfig.isEnabled() && siteNameABExperiment.getVariation() != Control) {
            add(SITE_NAME)
        }
        add(SITE_DESIGNS)
        add(DOMAINS)
        add(SITE_PREVIEW)
    }.toList()
```
Wdyt?

Reply:
I made this dependent in anticipation of the heading enhancement (make it dependent on the intent screen). But the flag itself doesn't need to be dependent, instead we can include the `INTENTS` with the site name flow variation. I've added this here: https://github.com/wordpress-mobile/WordPress-Android/pull/16259/commits/35edfb6a6057e8c323b14cd1a988c0d4a2ba7cc5 as a when expression, as an alternative (since this makes it nice to compare the lists, imo). Wdyt?

Reply:
Thank you for fixing this @mkevins. The solution with the `when` clause makes it even more clear 👍